src/app/shell/workshop-details/side-menu/actions/actions.component.html

```diff
@@ -1,11 +1,15 @@
 <div class="container" fxLayout="column" fxLayoutAlign="space-between center" fxLayoutGap='0.8rem'>
   <h4 *ngIf="workshop.price"> {{ workshop.price }} грн/ {{ workshop.isPerMonth ? 'місяць' : 'заняття' }} </h4>
   <h4 *ngIf="!workshop.price"> Безкоштовно </h4>
-  <a [routerLink]="['/create-application', workshop.id]" *ngIf="displayed">
-    <button mat-raised-button class="button button-application" fxLayout="row" fxLayoutAlign="center center">
+  <a [routerLink]="['/create-application', workshop.id]" *ngIf="displayed" >
+    <button   mat-raised-button class="button button-application" fxLayout="row" fxLayoutAlign="center center">
       Подати заявку
     </button>
-  </a>
+   </a>
+   <a *ngIf="!displayed" (click)="login()">
```

Comment:
According to the latest updates, we should hide this component (actions) for the provider role. Can you please apply these updates? Display this card only if a user is not registered or its role is a parent. Also, apply the same approach for redirecting to the registration page for the button "send messages" because it has the same conditions.